torch/_torch_docs.py

```diff
@@ -4625,6 +4624,53 @@ def merge_dicts(*dicts):
     torch.return_types.median(values=tensor([-0.3982,  0.2270,  0.2488,  0.4742]), indices=tensor([1, 4, 4, 3]))
 """.format(**single_dim_common))
 
+add_docstr(torch.nanmedian,
+           r"""
+nanmedian(input) -> Tensor
+
+This is a variant of :func:`torch.median` that "ignores" ``NaN`` values,
+computing the median as if ``NaN`` values in :attr:`input` did not exist.
+If all values are ``NaN`` then the median will be ``NaN``. See the 
+documentation for :func:`torch.median`.
+
+Args:
+    {input}
+
+Example::
+
+    >>> a = torch.tensor([1, float('nan'), 3, 2])
+    >>> a.median()
+    tensor(nan)
+    >>> a.nanmedian()
+    tensor(2.)
+
+.. function:: nanmedian(input, dim=-1, keepdim=False, *, out=None) -> (Tensor, LongTensor)
+
+This is a variant of :func:`torch.median` that "ignores" ``NaN`` values,
```

Comment:
Same notes apply here, except this needs to be careful to specify which part of input is being reduced. Also, the one-sentence description has to change to reflect that indices are returned, too:

"Reduces the specified dimensions of :attr:\`input\` to their medians, ignoring \`\`NaN\`\`s, and returning both the reduced tensor and a tensor with the original indices of the median elements."

Then:

"This function is identical to :func:`torch.median` when there are no NaN values in the reduced dimensions. When a reduced dimension has one or more NaN values, :func:`torch.median` will always reduce it to NaN, while this function will reduce it to the median of the non-NaN elements in that dimension. If all the elements of a reduced dimension are NaN then it will be reduced to NaN, too."

How does something like that sound?

Reply:
I think "Computes the median and its index along dimension :attr:`dim` of the :attr:`input` tensor, ignoring ``NaN``s." is concise and sufficient for the first sentence.